sapientml_core/templates/model_templates/model.py.jinja

```diff
@@ -22,21 +22,34 @@ model = {{ model_name }}({{ silent }}random_state=random_state_model, {{ params
 from sklearn.multioutput import MultiOutputRegressor
 
 model = MultiOutputRegressor(model)
+
 {% elif is_multioutput_classification %}
 from sklearn.multioutput import MultiOutputClassifier
 
 model = MultiOutputClassifier(model)
 {% endif %}
 {% set xgbclassifier = "XGBClassifier" %}
-{% if model_name == xgbclassifier %}
+{% if pipeline.task.target_columns|length == 1 %}
+
 from sklearn.preprocessing import LabelEncoder
 
-label_encoder = LabelEncoder()
-target_train = pd.DataFrame(label_encoder.fit_transform(target_train), columns=TARGET_COLUMNS)
-{% endif %}
-{% if pipeline.task.target_columns|length == 1 %}
+if target_train.select_dtypes(include=['object']).columns.any():
+        str_columns = target_train.select_dtypes(include=['object']).columns
```

Comment:
`str_columns` should be renamed as users this is target columns. For examples, `categorical_target_columns` and so on.